### DISCUSSION TITLE
Ability to add custom TipTap extensions

### DISCUSSION THREAD

**Discussion by jdjoshuadavison** [formfcw/directus-extension-flexible-editor#22] [Feature Request]
We have a few custom marks that we'd like to build and install, and I could see us wanting some custom nodes as well in the future. At the moment that will require us to modify the core of Flexible Editor.

It would be great if we could install custom TipTap extensions into a directory within the Flexible Editor bundle directory and automatically have Flexible Editor include them.

**Comment by invisible1988** [comment]
Hey Josh,

did you found a solution? I'm facing the similar problem right now.

Best

**Comment by djmtype** [comment]
That would be great if Flexible Editor supported an easy way to add TipTap extensions without having to fork and modify the code. 

Unless all of your content exists at the top-level – not nested – it's been a pain to group elements. 

It's confusing, adds duplication, and feels like inception. Unsure if this TipTap extension would solve that. https://github.com/topo-io/tiptap-extensions 

Anything is better than creating a whole lot of very specific, one-off, blocks just to add some sort of grid/flex layout.  

Adding a simple wrapper block, and assigning other blocks to it, isn't going to cut it.
